src/models/transaction.ts

```diff
@@ -177,18 +184,29 @@ class TransactionStore {
           blockNumber === 'latest' &&
           this._lastSyncedAccountsHash !== accountHash
         ) {
-          this._transactions = [];
+          const cahced = this._cache.get(`${accountHash}:${blockNumber}`);
+          if (cahced?.length) {
+            this._transactions = cahced;
+          } else {
+            this._transactions = [];
+          }
         }
 
         this._lastSyncedAccountsHash = accountHash;
         this._lastSyncedBlockNumber = blockNumber;
       });
 
-      return Indexer.instance.getTransactions(
+      const result = await Indexer.instance.getTransactions(
         accounts,
         blockNumber,
         app.providerId,
       );
+      await when(() => !Token.isLoading, {});
+      const parsed = result
+        .map(tx => parseTransaction(tx, accounts))
+        .filter(tx => !!tx.parsed);
+      this._cache.set(`${this._lastSyncedAccountsHash}:${blockNumber}`, parsed);
```

Comment:
I think it should be better to store key generation logic somewhere
```suggestion
      this._cache.set(this.generateKey(accounts, blockNumber), parsed);
```